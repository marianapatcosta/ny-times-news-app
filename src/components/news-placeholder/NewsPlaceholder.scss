$space: 0.75rem;

@mixin loading-item {
  background-color: $color-disabled;
  overflow: hidden;
  position: relative;
  opacity: 0.5;

  &:before {
    animation: loading 1s infinite linear;
    -webkit-animation: loading 1s infinite linear;
    background: linear-gradient(
      to right,
      $color-disabled,
      #ddd,
      $color-disabled
    );
    content: "";
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
  }
}

@keyframes loading {
  0% {
    transform: translateX(0);
  }
  50%,
  100% {
    transform: translateX(100%);
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    -o-transform: translateX(100%);
  }
}

.news-placeholder {
  background-color: $color-primary;
  box-shadow: 0 0.125rem 0.5rem rgba($color-black, 0.7);
  display: flex;
  flex-direction: column;
  max-width: 16rem;
  width: 100%;

  &__content {
    display: flex;
    flex-direction: column;
    flex: 1;
    padding: 1rem;
  }

  &__image {
    height: 11rem;
    @include loading-item;
  }

  &__header {
    height: 1.2rem;
    width: 100%;
    @include loading-item;
  }

  &__subheader {
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin-top: $space;
  }

  &__section-name {
    font-weight: 700;
    height: 0.9rem;
    width: 30%;
    @include loading-item;
  }

  &__details {
    height: 4rem;
    margin: $space 0;
    @include loading-item;
  }

  &__footer {
    height: 0.8rem;
    width: 50%;
    margin-top: auto;
    @include loading-item;
  }
}

@media only screen and (min-width: $xs) {
  .news-placeholder {
    max-width: 18.5rem;
  }
}
