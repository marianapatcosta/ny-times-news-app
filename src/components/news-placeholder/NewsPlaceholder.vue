<template>
  <li class="news-placeholder" data-cy="news-placeholder">
    <div class="news-placeholder__image" />
    <div class="news-placeholder__content">
      <div class="news-placeholder__header" />
      <div class="news-placeholder__subheader">
        <div class="news-placeholder__section-name" />
        <div class="news-placeholder__section-name" />
      </div>

      <div class="news-placeholder__main">
        <div class="news-placeholder__details" />
      </div>

      <div class="news-placeholder__footer" />
    </div>
  </li>
</template>

<script src="./NewsPlaceholder.js"></script>

<style scoped lang="scss" src="./NewsPlaceholder.scss"></style>
