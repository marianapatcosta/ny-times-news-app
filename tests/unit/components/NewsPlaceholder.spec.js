import { shallowMount } from "@vue/test-utils";
import NewsPlaceholder from "@/components/news-placeholder/NewsPlaceholder.vue";

describe("NewsPlaceholder.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(NewsPlaceholder);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render without errors and match snapshot renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
