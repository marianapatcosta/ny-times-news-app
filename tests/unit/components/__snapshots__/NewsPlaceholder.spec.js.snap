// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsPlaceholder.vue should render without errors and match snapshot renders correctly 1`] = `
<li
  class="news-placeholder"
  data-cy="news-placeholder"
>
  <div
    class="news-placeholder__image"
  />
  <div
    class="news-placeholder__content"
  >
    <div
      class="news-placeholder__header"
    />
    <div
      class="news-placeholder__subheader"
    >
      <div
        class="news-placeholder__section-name"
      />
      <div
        class="news-placeholder__section-name"
      />
    </div>
    <div
      class="news-placeholder__main"
    >
      <div
        class="news-placeholder__details"
      />
    </div>
    <div
      class="news-placeholder__footer"
    />
  </div>
</li>
`;
